mycpp/gc_dict.h

```diff
@@ -179,123 +175,103 @@ class Dict {
 };
 
 template <typename K, typename V>
-inline bool dict_contains(Dict<K, V>* haystack, K needle) {
-  return haystack->position_of_key(needle) != -1;
-}
-
-#if 0
-// mylib.NewDict() translates to this
-template <typename K, typename V>
-Dict<K, V>* NewDict() {
-  return Alloc<Dict<K, V>>();
+inline bool dict_contains(const Dict<K, V>* haystack, K needle) {
+  int pos = haystack->hash_and_probe(needle);
+  return pos != kNotFound && haystack->index_->items_[pos] >= 0;
 }
 
-template <typename K, typename V>
-Dict<K, V>* NewDict(std::initializer_list<K> keys,
-                    std::initializer_list<V> values) {
-  assert(keys.size() == values.size());
-  auto self = Alloc<Dict<K, V>>();
-  auto v = values.begin();  // This simulates a "zip" loop
-  for (auto key : keys) {
-    // note: calls reserve(), and maybe allocate
-    self->set(key, *v);
-    ++v;
-  }
-
-  return self;
-}
-#endif
-
 template <typename K, typename V>
 void Dict<K, V>::reserve(int n) {
```

Comment:
yeah I think this should be `int desired_entries`, and the contract is that you can fit that number of entries

So the caller should not have to call RoundCapacity()

Reply:
Sounds good